### PR TITLE
Test deeply nested data structures with canonical_json (fixes #26)

### LINKED ISSUE #26: We need to add test to validate that canonical_json handle nested dictionnary key sorting.

### PR DESCRIPTION
**Pull request by magopian**
Fixes #26

### CODE REVIEW

**Comment by magopian** [comment]
@almet r?

**Comment by almet** [comment]
Thanks, that looks great.

File: kinto_signer/tests/test_serializer.py [hunk 1]
```diff
@@ -63,3 +63,26 @@ def test_canonical_json_orders_object_keys():
...
+                'id': '1',
+                'e': 1,
+                'f': [2, 3, 1],
+                'g': {2: 2, 3: 3, 1: {
+                    'b': 'b', 'a': 'a', 'c': 'c'}}}},
+        'id': '1'}]
+    expected = (
+        '[{"a":{"a":"a","b":"b","c":{"a":"a","b":"b","c":["b","a","c"],'
+        '"d":{"a":"a","b":"b"},"e":1,"f":[2,3,1],"g":{'
+        '"1":{"a":"a","b":"b","c":"c"},"2":2,"3":3},"id":"1"}},"id":"1"}]')
```
Comment (almet) [inline kinto_signer/tests/test_serializer.py]: why is the id key present twice?

Comment (almet) [inline kinto_signer/tests/test_serializer.py]: Ah, disregard :-)

**Pull request merged**